scripts/post-processing/1-run_normalization.R

```diff
@@ -8,22 +8,30 @@
 #
 # Options:
 # '-d' : Path to gene matrix in tab delimited format, gene x sample with gene
-#        info as the first column. With gene info column labeled as 'gene' or 
+#        info as the first column. With gene info column labeled as 'gene' or
 #        'genes' (not case sensitive)",
 # '-a' : Normalization method to use. To run all methods use "all", to run more
 #        than one method, separate the names by a space. eg -a tmm log
+#        If scran is one of the methods chosen, it will be ran first because 
+#        samples with negative factor sizes will need to be removed. To make it
+#        parallel with the other methods, those same samples will be removed 
+#        from the dataset before other normalization methods are run. 
 #        Options:
 #           'scale'- scale and center each sample's data using base::scale
 #           'log'- just log transform alone
 #           'voom'- quantile normalization from the DEseq2::voom function
 #           'deseq2'- Use DEseq2 size factors
 #           'vsd'- Use DEseq2 varianceStabilizingTransformation function
 #           'tmm'- trimmed mean M values by edgeR
+#           'scran' - pooled CPM values
 #           'all'- run all the above methods
 # '-o' : Directory where you would like the output to go. Default is current
-#        directory. New folder will be created if the specified folder doesn't 
+#        directory. New folder will be created if the specified folder doesn't
 #        exist.
 # '-l' : Optional label for output files"
+# '-n' : Option to override scran negative factor warning, remove the problematic
+#        genes and normalize. Default is to not continue with scran normalization 
```

Comment:
Aren't you removing _samples_?

Reply:
Yeah, I'm not sure what my brain was thinking when I wrote this. Will change.